test/unit/src/common/containers/t_with-repository.js

```diff
@@ -0,0 +1,110 @@
+import React from 'react';
+import expect from 'expect';
+import { shallow } from 'enzyme';
+import configureMockStore from 'redux-mock-store';
+import thunk from 'redux-thunk';
+
+import withRepository from '../../../../../src/common/containers/with-repository';
+
+const middlewares = [ thunk ];
+const mockStore = configureMockStore(middlewares);
+
+// Dummy test component to wrap
+const Dummy = () => <span>Dummy</span>;
+
+import { SET_GITHUB_REPOSITORY } from '../../../../../src/common/actions/create-snap';
+
+describe('WithRepository', () => {
+  const PARAMS = {
+    owner: 'anowner',
+    name: 'aname'
+  };
+  const REPO = {
+    fullName: 'anowner/aname'
+  };
+
+  let store;
+
+  let wrapper;  // WithRepository rendered wrapper
+  let instance; // WithRepository wrapper instance
+  let wrapped;  // wrapped Dummy component
+
+  const setupDummy = (repository = REPO, params = PARAMS) => {
+    store = mockStore({ repository });
+
+    const DummyWithRepo = withRepository(Dummy);
+
+    // shallow render container component
+    const dummyWithRepo = shallow(<DummyWithRepo store={store} params={params}/>);
+
+    // get child components/instances from rendered component
+    wrapper = dummyWithRepo.first();
+    instance = wrapper.shallow().instance();
+    wrapped = wrapper.dive();
```

Comment:
Higher order components make components nested, so to properly test them with shallow rendering we need to dive into the children.